lms/djangoapps/discussion_api/api.py

```diff
@@ -244,6 +254,148 @@ def get_sorted_modules(category):
     }
 
 
+def _get_user_profile_dict(request, usernames):
+    """
+    Gets user profile details for a list of usernames and creates a dictionary with
+    profile details against username.
+
+    Parameters:
+
+        request: The django request object.
+        usernames: A string of comma separated usernames.
+
+    Returns:
+
+        A dict with username as key and user profile details as value.
+    """
+    username_profile_dict = {}
+    try:
+        request.GET = request.GET.copy()
+        request.GET['usernames'] = usernames
+        user_profile_details = AccountViewSet.as_view({'get': 'list'})(request).data
+    except Exception as ex:
+        raise ex
+
+    if isinstance(user_profile_details, list):
+        for user in user_profile_details:
+            username_profile_dict[user['username']] = user
+    else:
+        username_profile_dict[user_profile_details['username']] = user_profile_details
+    return username_profile_dict
+
+
+def _user_profile(user_profile):
+    """
+    Returns the user profile object with profile_image details.
+    """
```

Comment:
@jcdyer We don't have any more fields in user profile for now. Currently only the profile image details were required for each user participating in thread/comment either as author or endorser.

Are you suggesting we should add a set of fields that could be expected in future? or return the complete user account details we retrieved from accounts endpoint?
I believe the later would increase the response payload in size with unwanted data.